Library/Homebrew/formulary.rb

```diff
@@ -800,22 +805,30 @@ def self.loader_for(ref, from: nil)
   end
 
   def self.core_path(name)
-    CoreTap.instance.formula_dir/"#{name.to_s.downcase}.rb"
+    find_formula_in_tap(name.to_s.downcase, CoreTap.instance)
   end
 
   def self.core_alias_path(name)
     CoreTap.instance.alias_dir/name.to_s.downcase
   end
 
-  def self.tap_paths(name, taps = Dir[HOMEBREW_LIBRARY/"Taps/*/*/"])
+  def self.tap_paths(name, taps = Tap)
     name = name.to_s.downcase
     taps.map do |tap|
-      Pathname.glob([
-        "#{tap}Formula/#{name}.rb",
-        "#{tap}HomebrewFormula/#{name}.rb",
-        "#{tap}#{name}.rb",
-        "#{tap}Aliases/#{name}",
-      ]).find(&:file?)
-    end.compact
+      find_formula_in_tap(name, tap, default: false) ||
+        (tap.alias_dir/name) ||
+        (tap.formula_dir/"#{name}.rb")
```

Comment:
The last part of this will never evaluate.

Sounds like you are wannting to check `file?` on the middle part.